File: 2021/08/29/main.py
class Solution:
    def minPatches(self, nums: List[int], n: int) -> int:
        i, mx, r = 0, 1, 0
        while mx <= n:
            if i < len(nums) and nums[i] <= mx:
                mx += nums[i]
                i += 1    
            else:
                mx <<= 1
                r += 1
        return r

File: 2021/08/29/main.go
package main

func minPatches(nums []int, n int) int {
	i, mx, r := 0, 1, 0
	for mx <= n {
		if i < len(nums) && nums[i] <= mx {
			mx += nums[i]
			i++
		} else {
			mx <<= 1
			r++
		}
	}
	return r
}
